pkg/apis/networking/validation/validation_test.go

```diff
@@ -33,1022 +33,422 @@ import (
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Egress: []networking.NetworkPolicyEgressRule{
-					{
-						To: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-						Ports: []networking.NetworkPolicyPort{
-							{
-								Protocol: &protocolSCTP,
-								Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 33000},
-								EndPort:  &endPort,
-							},
-						},
-					},
-				},
-			},
-		},
-		"multiple invalid port ranges defined": {
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Egress: []networking.NetworkPolicyEgressRule{
-					{
-						To: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-						Ports: []networking.NetworkPolicyPort{
-							{
-								Protocol: &protocolUDP,
-								Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 35000},
-								EndPort:  &endPort,
-							},
-							{
-								Protocol: &protocolTCP,
-								EndPort:  &endPort,
-							},
-							{
-								Protocol: &protocolTCP,
-								Port:     &intstr.IntOrString{Type: intstr.String, StrVal: "https"},
-								EndPort:  &endPort,
-							},
-						},
-					},
-				},
-			},
-		},
-		"invalid endport range defined": {
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Egress: []networking.NetworkPolicyEgressRule{
-					{
-						To: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-						Ports: []networking.NetworkPolicyPort{
-							{
-								Protocol: nil,
-								Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 30000},
-								EndPort:  utilpointer.Int32Ptr(65537),
-							},
-						},
-					},
+		"invalid endport range defined": makeNetworkPolicyCustom(setEgressEmptyFirstElement, setEgressToEmptyFirstElement, setEgressToNamespaceSelector, func(networkPolicy *networking.NetworkPolicy) {
+			networkPolicy.Spec.Egress[0].Ports = []networking.NetworkPolicyPort{
+				{
+					Protocol: nil,
+					Port:     &port30000,
+					EndPort:  utilpointer.Int32Ptr(65537),
 				},
-			},
-		},
+			}
+		}),
 	}
 
 	// Error cases are not expected to pass validation.
 	for testName, networkPolicy := range errorCases {
-		if errs := ValidateNetworkPolicy(&networkPolicy); len(errs) == 0 {
+		if errs := ValidateNetworkPolicy(networkPolicy); len(errs) == 0 {
 			t.Errorf("Expected failure for test: %s", testName)
```

Comment:
I think we can add networkPolicy info in log. If faild, we can check it quickly.

Reply:
The networkPolicy struct has too many fields and again I think this would be a lot of noise in the test result. Would you print the whole struct or some specific fields?